### PR TITLE
Add impulse response functions to all rfunc

### PR DESCRIPTION
**Pull request by raoulcollenteur**
# Short Description

- Add impulse response functions to all rfunc
- Update init signatures
- Couple of small changes to docs

EDIT: ready for review.

# Checklist before PR can be merged:
- [x] closes issue #453
- [x] is documented
- [x] PEP8 compliant code
- [x] tests added / passed
- [x] Example Notebook (for new features)

### CODE REVIEW

**Comment by raoulcollenteur** [comment]
@dbrakenhoff this PR is ready for review.

File: pastas/decorators.py [hunk 1]
```diff
@@ -11,9 +14,9 @@ def set_use_numba(b: bool) -> None:
     USE_NUMBA = b
 
 
-def set_parameter(function):
+def set_parameter(function: Function) -> Decorator:
```
Comment (dbrakenhoff) [inline pastas/decorators.py:17]: I think a decorator returns a (modified) function

File: pastas/rfunc.py [hunk 2]
```diff
@@ -52,19 +52,34 @@
...
+    def __init__(
+        self, up: bool = True, meanstress: float = 1.0, cutoff: float = 0.999, **kwargs
+    ) -> None:
+        self.up = up
+        self.meanstress = meanstress
+        self.cutoff = cutoff
         self.kwargs = kwargs
 
-    def _set_init_parameter_settings(
+    def _update_rfunc_settings(
```
Comment (dbrakenhoff) [inline pastas/rfunc.py:63]: The response function notebook contains a reference to this function still.

File: pastas/rfunc.py [hunk 3]
```diff
@@ -158,34 +170,37 @@ def block(
...
+                response, can be obtained with get_t() method
+            p: array_like
+                array_like object with the values as floats representing the model
+                parameters.
 
-        Returns
-        -------
-        s: array_like
-            Array with the impulse response.
+            Returns
```
Comment (dbrakenhoff) [inline pastas/rfunc.py:189]: too many tabs?

**Review by dbrakenhoff — APPROVED** [review]
See comments

File: pastas/rfunc.py (hunk 3 above)
Comment (martinvonk) [reply pastas/rfunc.py:189]: fixed

File: pastas/rfunc.py (hunk 2 above)
Comment (raoulcollenteur) [reply pastas/rfunc.py:63]: Yes, this is for the next step when working on the documentation, merging these two notebooks eventually.

**Pull request merged**